apps/emqx/test/emqx_persistent_session_SUITE.erl

```diff
@@ -884,12 +903,16 @@ t_snabbkaffe_buffered_messages(Config) ->
            %% Make the resume init phase wait until the first message is delivered.
            ?force_ordering( #{ ?snk_kind := ps_worker_deliver },
                             #{ ?snk_kind := ps_resume_end }),
+           Parent = self(),
            spawn_link(fun() ->
                               ?block_until(#{?snk_kind := ps_marker_pendings_msgs}, infinity, 5000),
-                              publish(Topic, Payloads2, true)
+                              publish(Topic, Payloads2, true),
+                              Parent ! publish_done,
+                              ok
                       end),
            {ok, Client2} = emqtt:start_link([{clean_start, false} | EmqttOpts]),
```

Comment:
It's unlikely, but couldn't we have race condition, when publish process blocks _after_ session resume passed `ps_marker_pendings_msgs` stage?

Reply:
This race condition is exactly what I want to test.
I don't think the flakiness of this test is a real bug, though. It seems to be because snabbkaffe is terminated before all processes it has interacted with has passed all interaction points. That is what I'm trying to catch with the `publish_done` message.